godtools/ViewControllers/Tract/TractViewControllerPageMovements.swift

```diff
@@ -17,14 +17,12 @@ extension TractViewController {
             return
         }
         
-        let destinationViewPageId = dictionary["pageId"]
-        for (pageId, pageIndex) in self.pagesIds {
-            if pageId == destinationViewPageId {
-                self.currentPage = pageIndex
-                break
-            }
+        let pageListener = dictionary["pageListener"]
```

Comment:
I would do a `guard let` here too. If the listener is not present for some reason, then the app would crash on the next line.